pkg/sql/show_fingerprints.go

```diff
@@ -182,35 +219,54 @@ func (n *showFingerprintsNode) nextTenant(params runParams) (bool, error) {
 		return false, err
 	}
 
+	// We want to write a protected timestamp record at the earliest timestamp
+	// that the fingerprint query is going to read from. When fingerprinting
+	// revisions, this will be the specified start time.
+	tsToProtect := params.p.EvalContext().Txn.ReadTimestamp()
+	if n.options != nil && !n.options.startTimestamp.IsEmpty() {
+		if !n.options.startTimestamp.LessEq(tsToProtect) {
+			return false, errors.AssertionFailedf("start timestamp %s is greater than the end timestamp %s",
+				n.options.startTimestamp.String(), tsToProtect.String())
```

Comment:
I don't think an AsserionFailedf is quite right here since it is the result of user input. Maybe ` pgerror.Newf(pgcode.InvalidParameterValue,`